src/components/include/protocol_handler/protocol_handler.h

```diff
@@ -138,7 +138,7 @@ class ProtocolHandler {
    * generated_session_id is 0.
    */
   virtual void NotifySessionStarted(
-      const SessionContext& context,
+      SessionContext& context,
```

Comment:
unfortunately this is not allowed on a public method.

to change this parameter to non-const we need to mark this method as DEPRECATED and define a new method with the new signature.

Reply:
@iCollin sorry for that. I declared a new function for that and deprecated the old one.
Note, that I have to make some extra tricks with casts in unit tests to avoid ambiguous call errors.
Fixed in https://github.com/smartdevicelink/sdl_core/pull/3647/commits/77231fce35b028d5691fd42efaff4e155a47be3b